33-lifecycle-lab_v16/src/01-cps.jsx

```diff
@@ -9,12 +9,35 @@
 import React from 'react';
 
 export default class Cps extends React.Component{
+  constructor(props){
+    super(props);
+    this.state = { rate  :0 , count : 0 };
+  }
+  
+  
+  componentWillMount = () => {
```

Comment:
Don't use componentWillMount. Ever. Everything you want to write here is better written in the constructor or in componentDidMount.
(If it uses the DOM it should be in componentDidMount. Otherwise in constructor)